pkg/agent/informant.go

```diff
@@ -1091,3 +1175,93 @@ func (s *InformantServer) Upscale(ctx context.Context, logger *zap.Logger, to ap
 	logger.Info("Received successful upscale result")
 	return nil
 }
+
+func (s *InformantServer) MonitorHealthCheck(ctx context.Context, logger *zap.Logger) error {
+	tx, rx := util.NewSingleSignalPair[*MonitorResult]()
+	err := s.dispatcher.Call(
+		ctx,
+		tx,
+		api.HealthCheck{},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Wait for result
+	timeout := time.Second * time.Duration(s.runner.global.config.Monitor.ResponseTimeoutSeconds)
+	select {
+	case res := <-rx.Recv():
+		// A nil pointer means a monitor error occured
+		if res != nil {
+			return nil
+		} else {
+			return fmt.Errorf("monitor experienced an internal error")
+		}
+	case <-time.After(timeout):
+		return fmt.Errorf("timed out waiting %v for monitor response", timeout)
```

Comment:
`time.After` spawns a goroutine that isn't cleaned up until the time expires (IIRC). For this, it'd be better to use something like `timer := time.NewTimer(...); defer timer.Stop()`